src/components/RequestPanel/RequestPanel.js

```diff
@@ -0,0 +1,291 @@
+      .filter(req => req.ensureConfirmation)
+
+    const { confirmed, errorProcessing, failed, processed } = progress
+    const status = (request, index) => {
+      // Request processing failed
+      if (errorProcessing !== -1 && index >= errorProcessing) {
+        return REQUEST_STATUS_PROCESSING_FAILED
+      }
+
+      // Request confirmation failed
+      if (failed !== -1 && index >= failed) {
+        return REQUEST_STATUS_FAILED
+      }
+
+      if (request.ensureConfirmation) {
+        const confirmedIndex = requiredConfirmedRequests.findIndex(
+          req => req.index === index
+        )
+
+        // Request confirmed
+        if (confirmedIndex < confirmed) {
+          return REQUEST_STATUS_CONFIRMED
+        }
+
+        // Request pending
+        if (index < processed) {
+          return REQUEST_STATUS_PENDING
+        }
+      }
+
+      // Request processed
+      if (index < processed) {
+        return REQUEST_STATUS_PROCESSED
+      }
+
+      // Transaction not processed
+      return REQUEST_STATUS_UPCOMING
+    }
+
+    return requests.map((request, index) => status(request, index))
+  }, [progress, requests])
+
+  const allSuccess = useMemo(() => {
+    const { confirmed, processed } = progress
+
+    const requiredConfirmedRequests = requests.filter(
+      request => request.ensureConfirmation
+    ).length
+
+    const allProcessed = processed > 0 && processed === requests.length
+    const allConfirmed = confirmed === requiredConfirmedRequests
+    return allProcessed && allConfirmed
+  }, [progress, requests])
+
+  const maxAttemptsReached = attempts >= MAX_ATTEMPTS
+
+  // Create requests
+  useEffect(
+    () => {
+      if (!progress.requesting || maxAttemptsReached) {
+        return
+      }
+
+      if (attempts === 0) {
+        setProgress(({ requesting }) => ({
+          ...INITIAL_STATE,
+          requesting,
+        }))
+      } else {
+        setProgress(progress => ({
+          ...progress,
+          errorProcessing: -1,
+          failed: -1,
+        }))
+      }
+
+      let cancelled = false
+
+      const createRequests = async () => {
+        const filteredRequests = requests.slice(progress.processed)
+
+        for (const request of filteredRequests) {
+          if (cancelled) {
+            break
+          }
+
+          try {
+            const requestProcess = await processRequest(request)
+
+            if (!cancelled) {
+              if (request.isTx && request.ensureConfirmation) {
+                await ensureRequestConfirmed(requestProcess)
+
+                // Execute callback function if required
+                if (typeof request?.callback === 'function') {
+                  request.callback()
+                }
+              }
+            }
+          } catch (err) {
+            console.error(`Error when executing: ${request.description}:`, err)
+            captureException(err)
+
+            if (requests.length > 1) {
```

Comment:
Perhaps we can always just re-throw here to force us to exit.

Using the exception seems natural as it's a native "escape hatch" to normal control flow.